gaia-component.js

```diff
@@ -147,6 +83,26 @@ var base = {
     },
 
     /**
+     * A convenient method for triggering l10n for component's shadow DOM.
+     */
+    setupShadowL10n: function() {
+      if (!document.l10n) { return; }
+      this.onDOMRetranslated = this.localizeShadow.bind(null, this.shadowRoot);
+      document.l10n.ready.then(() => {
+        document.addEventListener('DOMRetranslated', this.onDOMRetranslated);
+        this.localizeShadow(this.shadowRoot);
+      });
+    },
+
+    /**
+     * Localizes the shadowRoot subtree.
+     */
+    localizeShadow: function(shadowRoot) {
+      if (!document.l10n) { return; }
+      document.l10n.translateFragment(shadowRoot);
+    },
```

Comment:
Should we doucment that this function can be overriden on the prototype? Although the current logic would be that it's never run if `l10n.js` is not being used. Is this is 'by design' I'm OK with that. Depends how important you think it is that third-parties can use the component without `l10n.js` :)